editor/src/components/canvas/remix/utopia-remix-root-component.tsx

```diff
@@ -190,6 +213,45 @@ function useGetRoutes() {
   ])
 }
 
+// Super hacky way of calling the `fetch` function from `server.js` purely to get the context
+// that it provides, so that we can then provide that to the server function
+async function getContextFromCustomServer(
```

Comment:
something about this doesn't sit right with me, but I don't yet have anything better to offer

Reply:
Something about this "super hacky" horror show doesn't sit right with you? 😁 **ALL** of it doesn't sit right with me!